autoemulate/simulations/seir.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import numpy as np
+import torch
+from scipy.integrate import solve_ivp
+
+from autoemulate.core.types import NumpyLike, TensorLike
+from autoemulate.simulations.base import Simulator
+
+
+def simulate_seir_epidemic(
+    x: NumpyLike,
+    N: int = 1000,
+    I0: int = 1,
+    E0: int = 0,
+) -> float:
+    """
+    Simulate an epidemic using the SEIR model.
+
+    Parameters
+    ----------
+    x : NumpyLike
+        SEIR parameters [beta, gamma, sigma].
+    N : int
+        Total population.
+    I0 : int
+        Initial infected.
+    E0 : int
+        Initial exposed.
+
+    Returns
+    -------
+    peak_infection_rate : float
+        Peak infection fraction I_max / N.
+    """
+    if len(x) != 3:
+        raise ValueError(f"Expected 3 parameters [beta, gamma, sigma], got {len(x)}")
+
+    beta, gamma, sigma = x
+
+    S0 = N - I0 - E0
+    R0 = 0
+    t_span = (0.0, 160.0)
+    y0 = [S0, E0, I0, R0]
+
+    def seir_model(t, y, N, beta, gamma, sigma):
+        S, E, I, R = y
```

Comment:
Similarly, it complains about ambiguous (e.g., one letter) variable names - this is a way to skip this check (alternative is to rename the variables)

```suggestion
        S, E, I, R = y  # noqa: E741
```